bluetooth_mesh/models/models.py

```diff
@@ -0,0 +1,1828 @@
+"""
+This module implements mesh models, both clients and servers.
+"""
+import inspect
+import itertools
+from functools import partial
+from typing import Any, Dict, Iterable, NamedTuple, Optional, Sequence, Tuple, Type
+
+from construct import BitStruct
+
+from bluetooth_mesh.crypto import ApplicationKey, NetworkKey
+from bluetooth_mesh.messages.config import (
+    CompositionData,
+    ConfigOpcode,
+    PublishFriendshipCredentialsFlag,
+    SecureNetworkBeacon,
+    StatusCode,
+)
+from bluetooth_mesh.messages.generic.level import GenericLevelOpcode
+from bluetooth_mesh.messages.generic.light.ctl import LightCTLOpcode
+from bluetooth_mesh.messages.generic.light.lightness import LightLightnessOpcode
+from bluetooth_mesh.messages.generic.onoff import GenericOnOffOpcode
+from bluetooth_mesh.messages.health import HealthOpcode
+from bluetooth_mesh.messages.scene import SceneMessageOpcode
+from bluetooth_mesh.messages.sensor import SensorOpcode
+from bluetooth_mesh.messages.silvair.debug import DebugOpcode, DebugSubOpcode
+from bluetooth_mesh.messages.silvair.gateway_config_server import (
+    DhcpFlag,
+    GatewayConfigServerOpcode,
+    GatewayConfigServerSubOpcode,
+)
+from bluetooth_mesh.messages.silvair.network_diagnostic_server import (
+    NetworkDiagnosticServerOpcode,
+    NetworkDiagnosticSetupServerOpcode,
+)
+from bluetooth_mesh.models.base import Model
+from bluetooth_mesh.utils import ModelOperationError, ProgressCallback
+
+__all__ = [
+    "ConfigServer",
+    "ConfigClient",
+    "HealthServer",
+    "HealthClient",
+    "DebugServer",
+    "DebugClient",
+    "NetworkDiagnosticClient",
+    "NetworkDiagnosticSetupClient",
+    "GenericOnOffServer",
+    "GenericOnOffClient",
+    "SceneClient",
+    "GenericLevelClient",
+    "SensorClient",
+    "LightLightnessClient",
+    "LightLightnessServer",
+    "LightCTLClient",
+    "GatewayConfigServer",
+    "GatewayConfigClient",
+]
```

Comment:
This module is too long. Please split it logically into groups of modes.

Reply:
Ok, but not now.

Reply:
Understandable. Should I create an issue so that it doesn't get lost?

Reply:
https://silvair.atlassian.net/browse/SG-392